core/primitives/src/transaction.rs

```diff
@@ -480,7 +480,8 @@ mod tests {
 
     #[test]
     fn test_verify_transaction() {
-        let signer = InMemorySigner::from_random("test".parse().unwrap(), KeyType::ED25519);
+        let signer: Signer =
+            InMemorySigner::from_random("test".parse().unwrap(), KeyType::ED25519).into();
```

Comment:
Instead of into() would it make sense to have the constructors of InMemorySigner and EmptySigner return Signer? If access to the inner enum value is needed you can expose that via new methods. Same for Signer and ValidatorSigner.

Reply:
I thought about it but `account_id` field does not exist in `EmptySigner`.
We could return empty string or `unimplemented` for `EmptySigner` but this code path should never be called anyway.
I have no strong opinion, looks like doing what you said would simplify this PR, I will try it out, thanks!

Reply:
I've spent a while trying to get rid of `into()`, but it was not straightforward. There are some places that use members like `secret_key`. I would rather do it as a follow up PR.

Reply:
Will do it as a separate issue: https://github.com/near/nearcore/issues/11531